project-base/storefront/components/Blocks/Product/ProductsList/ProductListItemImage.tsx

```diff
@@ -0,0 +1,56 @@
+import { ProductItemProps } from './ProductListItem';
+import { Image } from 'components/Basic/Image/Image';
+import { TIDs } from 'cypress/tids';
+import dynamic from 'next/dynamic';
+import { twJoin } from 'tailwind-merge';
+import { twMergeCustom } from 'utils/twMerge';
+
+const ProductFlags = dynamic(() => import('../ProductFlags').then((component) => component.ProductFlags));
+
+type ProductListItemImageProps = {
+    size: ProductItemProps['size'];
+    product: ProductItemProps['product'];
+    visibleItemsConfig: ProductItemProps['visibleItemsConfig'];
+};
+
+export const ProductListItemImage: FC<ProductListItemImageProps> = ({ product, visibleItemsConfig, size }) => {
+    const imageSize = (() => {
+        switch (size) {
+            case 'big':
+                return 180;
+            case 'medium':
+                return 150;
+            case 'small':
+                return 94;
+            default:
+                return 180;
+        }
+    })();
+
+    return (
+        <div
+            className={twMergeCustom('relative flex items-center justify-center mx-auto w-full')}
+            tid={TIDs.product_list_item_image}
+        >
+            <Image
+                alt={product.mainImage?.name || product.fullName}
+                draggable={false}
+                height={imageSize}
+                src={product.mainImage?.url}
+                width={imageSize}
+                className={twJoin(
+                    'max-h-full object-contain',
+                    size === 'small' && 'h-[94px] w-[94px]',
+                    size === 'medium' && 'h-[150px] w-[150px]',
+                    size === 'big' && 'h-[180px] w-[180px]',
```

Comment:
Cannot you just use `h-[${imageSize}]` ? Is that not optimal?

Reply:
Good point, fixed